eureka/S6_planet_spectra/s6_spectra.py

```diff
@@ -0,0 +1,281 @@
+                            new_file.write(line)
+                        else:
+                            line_segs = line.strip().split()
+                            if line_segs[0]=='inputdir':
+                                new_file.write(line_segs[0]+'\t\t/'+meta.inputdir+'\t'+' '.join(line_segs[2:])+'\n')
+                            else:
+                                new_file.write(line)
+            
+            # Get the wavelength values
+            wavelengths = np.mean(np.append(meta.wave_low.reshape(1,-1), meta.wave_hi.reshape(1,-1), axis=0), axis=0)
+            wave_errs = (meta.wave_hi-meta.wave_low)/2
+
+            fit_methods = meta.fit_method.strip('[').strip(']').strip().split(',')
+
+            # Read in S5 fitted values
+            if meta.sharedp:
+                medians, errs, ylabel = parse_s5_saves(meta, fit_methods, channel_key='shared')
+            else:
+                medians = []
+                errs = []
+                for channel in range(meta.nspecchan):
+                    median, err, ylabel = parse_s5_saves(meta, fit_methods, channel_key=f'ch{channel}')
+                    medians.append(median[0])
+                    errs.append(np.array(err).reshape(-1))
+                medians = np.array(medians).reshape(-1)
+                errs = np.array(errs).swapaxes(0,1)
+                if np.all(errs==None):
+                    errs = None
+
+            if meta.model_spectrum is not None:
+                model_x, model_y = np.loadtxt(os.path.join(meta.topdir, *meta.model_spectrum.split(os.sep)), delimiter=',').T
+            else:
+                model_x = None
+                model_y = None
+
+            # Make the spectrum plot
+            plots.plot_spectrum(meta, wavelengths, medians, errs, wave_errs, model_x, model_y, ylabel)
+            
+            # Calculate total time
+            total = (time_pkg.time() - t0) / 60.
+            log.writelog('\nTotal time (min): ' + str(np.round(total, 2)))
+            
+            # Save results
+            log.writelog('Saving results')
+            me.saveevent(meta, meta.outputdir + 'S6_' + meta.eventlabel + "_Meta_Save", save=[])
+            
+            log.closelog()
+    
+    return meta
+
+def parse_s5_saves(meta, fit_methods, channel_key='shared'):
+    for fitter in fit_methods:
+        if fitter in ['dynesty', 'emcee']:
+            fname = f'S5_{fitter}_samples_{channel_key}.csv'
+            samples = pd.read_csv(meta.inputdir+fname, escapechar='#', skipinitialspace=True)
+            # FINDME: kludge to decide between transmission and emission spectra for now
+            if 'fp' in samples.keys():
+                keys = [key for key in samples.keys() if 'fp' in key]
+                ylabel=r'$F_{\rm p}/F_{\rm *}$ (ppm)'
+                scalar = 1e6
+            else:
+                keys = [key for key in samples.keys() if 'rp' in key]
+                ylabel=r'$R_{\rm p}/R_{\rm *}$'
```

Comment:
I guess setting what units to plot in would be a cleaner way of choosing between fp/fs and rp/rs than what I have too. If it can take a list, then it will even kind of work for phase curves (or at least be closer to working)